test/test-utils.js

```diff
@@ -1,23 +1,54 @@
 const fs = require('fs');
-const path = require('path');
-const {execSync} = require('child_process');
+const util = require('util');
+const glob = require('glob');
+const exec = util.promisify(require('child_process').exec);
+
+const STRIP_CHUNKHASH = /([\w\-]+\-)[a-z0-9]{8}(\.js)/g;
+const STRIP_REV = /\?rev=\w+/gm;
+const STRIP_WHITESPACE = /((\s+$)|((\\r\\n)|(\\n)))/gm;
+const UTF8_FRIENDLY_EXTS = ['css', 'html', 'js', 'jsx', 'ts', 'tsx', 'svelte', 'svg', 'vue']; // only read non-binary files (add more exts here as needed)
 
 /** setup for /tests/build/* */
-function setupBuildTest(cwd) {
-  execSync('yarn testbuild', {cwd});
+async function setupBuildTest(cwd) {
+  const results = await exec('yarn testbuild', {cwd});
+  return results;
 }
 exports.setupBuildTest = setupBuildTest;
 
-/** take an array of files, return contents of files (won’t read entire directory for performance) */
-function readFiles(files, {cwd}) {
-  if (!cwd) throw new Error(`cwd option missing, ex: readFiles(files, { cwd: __dirname })`);
+/** take a folder of files, return contents of files (won’t read entire directory for performance) */
+async function readFiles(directory, {ignore} = {}) {
```

Comment:
As stated above, this now just reads all UTF-8-friendly files in a directory. But if you have a test file where a particular file is too large, you can always `ignore` it from `glob()`.

In other words, this assumes test directories are tiny by default, but if that assumption isn’t correct you can always ignore individual files from being read into memory.